transformer_engine/pytorch/tensor/float8_tensor.py

```diff
@@ -680,10 +682,12 @@ def to_float8(
         fp8_meta_forward: bool = True,
         fp8_meta_index: Optional[int] = None,
         fp8_dtype: TE_DType = TE_DType.kFloat8E4M3,
+        data: Optional[torch.Tensor] = None,
         scale: Optional[torch.Tensor] = None,
         amax: Optional[torch.Tensor] = None,
         scale_inv: Optional[torch.Tensor] = None,
         with_transpose_cache: bool = False,
+        data_transpose: Optional[torch.Tensor] = None,
```

Comment:
- The `data` kwarg allows us to easily initialize `Float8Tensor`s that use the UB workspace buffer
- The `data_transpose` kwarg is added for completeness